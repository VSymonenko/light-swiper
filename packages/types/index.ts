export type Effect = 'slide' | 'fade';

export interface Pagination {
  clickable: boolean;
  activeClass: string;
}

export interface Navigation {
  icons?: {
    next: string;
    previous: string;
  },
  calibration?: number;
}

export interface Breakpoints {
  [key: number]: {
    slidesPerView?: number;
    spaceBetweenSlides?: number;
  }
}

export interface SwiperOptions {
  netflix?: boolean;
  lock?: boolean;
  effect?: Effect;
  sensitivity?: number;
  activeClass?: string;
  width?: number;
  height?: number;
  loop?: boolean;
  navigation?: Navigation | boolean;
  pagination?: Pagination | boolean;
  slidesPerView?: number;
  spaceBetweenSlides?: number;
  speed?: number;
  breakpoints?: Breakpoints;
  index?: number;
  template?: (context: SlideItem) => Element | string;
}

export type CSSSelector = string;

export type CSSSelectorType = 'class' | 'id';

export type Slide = string | SlideItem | SlideNode;

export type SlideNode = HTMLElementTagNameMap[keyof HTMLElementTagNameMap];

export type SlideType = 'video' | 'picture';

// includes attributes for <source>
// eslint-disable-next-line max-len
// https://html.spec.whatwg.org/multipage/embedded-content.html#the-source-element
export interface SlideItem {
  type: SlideType;
  source: Source;
  image: Image;
  link: string;
}

export interface Source {
  src: string;
  type: string;
  srcset: string;
  sizes: string;
  media: string;
}

export interface Image {
  src: string;
  alt: string;
  srcset: string;
  sizes: string;
}

export interface ISwiper extends SwiperOptions {
}
