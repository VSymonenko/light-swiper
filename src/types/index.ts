export type Effect = 'slide' | 'fade';

export interface Pagination {
  clickable: boolean;
  activeClass: string;
}

export interface Navigation {
  icons?: {
    next: string;
    previous: string;
  },
  calibration?: number;
}

export interface Breakpoints {
  [key: number]: {
    slidesPerView?: number;
    spaceBetweenSlides?: number;
  }
}

export interface SwiperOptions {
  lock?: boolean;
  anchor?: string;
  effect?: Effect;
  sensitivity?: number;
  activeClass?: string;
  fullWidth?: boolean;
  fullHeight?: boolean;
  width?: number;
  height?: number;
  loop?: boolean;
  isMobile?: boolean;
  navigation?: Navigation;
  pagination?: Pagination;
  slidesPerView?: number;
  spaceBetweenSlides?: number;
  speed?: number
  breakpoints?: Breakpoints;
}

export type CSSSelector = string;

export type CSSSelectorType = 'class' | 'id';

export interface ISwiper extends SwiperOptions {
}
