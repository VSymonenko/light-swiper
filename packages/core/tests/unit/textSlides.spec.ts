import Swiper from '@/swiper';
import { Slide } from '@/types';

let swiper: Swiper;
const el = document.createElement('div');
const classSelector = 'swiper-container';
el.className = classSelector;
document.body.appendChild(el);

beforeEach(() => {
  const slides: Slide[] = ['Slide 1', 'Slide 2', 'Slide 3'];
  swiper = new Swiper(`.${classSelector}`, slides);
});

test('slide property should be text types', () => {
  expect(swiper.slides).toEqual(
    expect.arrayContaining(['Slide 1', 'Slide 2', 'Slide 3']),
  );
});

test('should be changed slides', () => {
  const newSlides = ['New slide 1', 'New slide 2'];
  swiper.slides = newSlides;

  expect(swiper.slides).toEqual(
    expect.arrayContaining(['New slide 1', 'New slide 2']),
  );
});

test('should be render text nodes', () => {
  expect(swiper.slideWrapper).toMatchSnapshot();
});
