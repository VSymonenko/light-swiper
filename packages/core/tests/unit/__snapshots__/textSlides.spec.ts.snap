// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be render text nodes 1`] = `
<div
  class="swiper-wrapper"
/>
`;
